src/Native/Unix/System.Native/pal_process.cpp

```diff
@@ -450,34 +450,46 @@ extern "C" void SystemNative_SysLog(SysLogPriority priority, const char* message
     syslog(static_cast<int>(priority), message, arg1);
 }
 
-extern "C" int32_t SystemNative_WaitIdExitedNoHang(int32_t pid, int32_t* exitCode, int32_t keepWaitable)
+extern "C" int32_t SystemNative_WaitIdAnyExitedNoHangNoWait()
 {
-    assert(exitCode != nullptr);
-
     siginfo_t siginfo;
     int32_t result;
-    idtype_t idtype = pid == -1 ? P_ALL : P_PID;
-    int options = WEXITED | WNOHANG;
-    if (keepWaitable != 0)
-    {
-        options |= WNOWAIT;
-    }
-    while (CheckInterrupted(result = waitid(idtype, static_cast<id_t>(pid), &siginfo, options)));
-    if (idtype == P_ALL && result == -1 && errno == ECHILD)
+    while (CheckInterrupted(result = waitid(P_ALL, 0, &siginfo, WEXITED | WNOHANG | WNOWAIT)));
```

Comment:
Did you mean to change the pid from -1 to 0?  From the man page, that would seem to restrict this to only be for child processes in the same group.

Reply:
Linux man page: https://linux.die.net/man/3/waitid

> If idtype is P_ALL, waitid() shall wait for any children and id is ignored.

http://pubs.opengroup.org/onlinepubs/9699919799/functions/waitid.html has the same line.

Reply:
Ok, thanks.